core/src/main/scala/kafka/log/LogCleanerManager.scala

```diff
@@ -511,6 +513,27 @@ private[log] class LogCleanerManager(val logDirs: Seq[File],
       uncleanablePartitions.get(log.parentDir).exists(partitions => partitions.contains(topicPartition))
     }
   }
+
+  def maintainUncleanablePartitions(): Unit = {
```

Comment:
Thinking a bit more. LogCleanerManager.grabFilthiestCompactedLog() checks the current logs periodically. Could we just piggyback the maintenance of uncleanablePartitions there instead of introducing a separate periodic task?

Reply:
@junrao, I tried to put the logic in LogCleaner.doWork at the beginning (which I feel is similar to LogCleanerManager.grabFilthiestCompactedLog) but then reverted the change simply because maintenance work performed more frequently. If the overhead of frequent maintenance is not a concern (I assume the maintenance task is cheap), we can definitely do this.

Reply:
Pushed a new iteration